ICMP: Introduce ICMP type name in ICMPField (#30330)
cilium/cilium: Documentation/operations/upgrade.rst

  met (e.g: NodePort service implementation in BPF is disabled or socket load-balancing
  is disabled), it will fail to initialize and will log an error instead of silently
  fall back to iptables based masquerading.

> * The Go types for Network policies have changed to accept both integers and strings in
>   the ICMP protocol type. Specifically, the ICMP ``type`` field can be either

Reviewer: ```suggestion
* The ICMP ``type`` field in Network Policy now can be either
```
Or something similar - I think the upgrade notes should highlight the change in the CRD (user-facing) rather than in Go types (implementation detail).
